packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -742,12 +776,53 @@ function getHealthCheckCommand(hc: HealthCheck): string[] {
   return hcCommand.concat(cmd);
 }
 
-function renderResourceRequirements(gpuCount: number): CfnTaskDefinition.ResourceRequirementProperty[] | undefined {
-  if (gpuCount === 0) { return undefined; }
-  return [{
-    type: 'GPU',
-    value: gpuCount.toString(),
-  }];
+/**
+ * The type and amount of a resource to assign to a container.
+ */
+export interface ResourceRequirement {
+  /**
+   * The type of resource to assign to a container.
+   */
+  readonly type: ResourceRequirementType,
+
+  /**
+   * The value for the specified resource type.
+   */
+  readonly value: string,
+}
+
+/** Type of resource to assign to a container. */
+export enum ResourceRequirementType {
+  /**
+   * GPU.
+   */
+  GPU = 'GPU',
+  /**
+   * InferenceAccelerator.
+   */
+  INFERENCE_ACCELERATOR = 'InferenceAccelerator'
+}
+
+function renderResourceRequirements(gpuCount: number = 0, resourceRequirements: ResourceRequirement[] = []):
+CfnTaskDefinition.ResourceRequirementProperty[] | undefined {
+  if (resourceRequirements.length > 0) {
+    const ret = [];
+
+    for (const resource of resourceRequirements) {
+      ret.push({
+        type: ResourceRequirementType.INFERENCEACCELERATOR,
+        value: resource.value,
+      });
+    }
+    return ret;
+  }
+  if (gpuCount > 0) {
+    return [{
+      type: ResourceRequirementType.GPU,
+      value: gpuCount.toString(),
+    }];
+  }
```

Comment:
If a user set both `gpuCount` and `resourceRequirements `, would they be surprised to see their `gpuCount` config is not rendered and there's no error either 🤔

Reply:
I see your point, need to check if using both is a valid case. Will otherwise add the error condition.